tests/framework/extensions/provisioning/ssh.go

```diff
@@ -8,38 +8,83 @@ import (
-	checkCPU    = "CheckCPU"
+	cpuUsageVar                                            = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
+	checkCPU                 provisioninginput.SSHTestCase = "CheckCPU"
+	checkCPUCommand                                        = "ps -C agent -o %cpu --no-header"
+	nodeReboot               provisioninginput.SSHTestCase = "NodeReboot"
+	activeState                                            = "active"
+	runningState                                           = "running"
+	machineSteveResourceType                               = "cluster.x-k8s.io.machine"
+	fleetNamespace                                         = "fleet-default"
 )
 
 // CallSSHTestByName tests the ssh tests specified in the provisioninginput config clusterSSHTests field.
 // For example CheckCPU checks the cpu usage of the cluster agent. If the usage is too high the func will return a warning.
-func CallSSHTestByName(testName string, node *nodes.Node) error {
-	switch testName {
+func CallSSHTestByName(testCase provisioninginput.SSHTestCase, node *nodes.Node, client *rancher.Client, clusterObject *steveV1.SteveAPIObject, clusterID string, machineName string) error {
+	switch testCase {
 	case checkCPU:
-		command := "ps -C agent -o %cpu --no-header"
-		output, err := node.ExecuteCommand(command)
+		logrus.Infof("Running CheckCPU test on node %s", node.PublicIPAddress)
+		output, err := node.ExecuteCommand(checkCPUCommand)
 		if err != nil {
 			return err
 		}
-		strOutput := output[:strings.IndexByte(output, '\n')]
-		logrus.Info("CheckCPU test on node " + node.PublicIPAddress + " | Cluster agent cpu usage is: " + strOutput + "%")
+		str_output := output[:strings.IndexByte(output, '\n')]
+		logrus.Info("CheckCPU test on node " + node.PublicIPAddress + " | Cluster agent cpu usage is: " + str_output + "%")
 
-		outputInt, err := strconv.ParseFloat(strings.TrimSpace(strOutput), 32)
-		if outputInt > cpuUsageVar {
-			logrus.Warn("Cluster agent cpu usage is too high on node" + node.PublicIPAddress + " | Current cpu usage is: " + strOutput + "%")
+		output_int, err := strconv.ParseFloat(strings.TrimSpace(str_output), 32)
+		if output_int > cpuUsageVar {
+			logrus.Warn("Cluster agent cpu usage is too high on node" + node.PublicIPAddress + " | Current cpu usage is: " + str_output + "%")
 		}
 		if err != nil {
 			return err
 		}
+	case nodeReboot:
+		logrus.Infof("Running NodeReboot test on node %s", node.PublicIPAddress)
+		command := "sudo reboot"
+		_, err := node.ExecuteCommand(command)
+		if err != nil && !errors.Is(err, &ssh.ExitMissingError{}) {
+			return err
+		}
+		// Verify machine shuts down within 30 seconds, shutting down should not take longer than that
+		err = wait.Poll(1*time.Second, defaults.ThirtySecondTimeout, func() (bool, error) {
```

Comment:
So the poll now waits for the machine to go in a none running state? This is much better